### PR TITLE
Allow the user to define fallbacks for locales and languages

### PR DESCRIPTION
**Pull request by oliversalzburg**
Using the `fallbacks` map in the options array, the user can define which locales or languages should be mapped to which locale or language.

As an example, we use this fallback system like so:

```js
i18n.configure( {
	locales        : [ "en-US", "de-DE" ],
	fallbacks      : { "en" : "en-US", "de" : "de-DE" },
	defaultLocale  : "en-US"
} );
```

This allows us to be explicit in our locale definition (using "de-DE" instead of "de"), but map other regions to the same language.

The current fallback mechanism in `guessLanguage` does not provide the same kind of functionality or flexibility.

Please note that the tests currently forcibly remove the i18n instance from the cache to allow proper recreation of the loaded locales. This would be fixed by #130

### CODE REVIEW

**Comment by themouette** [comment]
any news on this?

**Comment by Mattijah** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -40,6 +40,9 @@ Minimal example, just setup two locales and a project specific directory
 	    // setup some locales - other locales default to en silently
 	    locales:['en', 'de'],
 
+		// fall back from Dutch to German
+	    fallbacks:['nl':'de'],
```
Comment (Mattijah) [inline README.md]: `fallbacks` should be probably an object not an array

Comment (Mattijah) [inline README.md]: `fallbacks` should be probably an object not an array

**Comment by oliversalzburg** [comment]
@Mattijah Should be fixed

**Comment by Mattijah** [comment]
@oliversalzburg great. By the way wouldn't be better if the fallbacks were supporting across the whole API? Look for instance at this https://github.com/Mattijah/i18n-node/commit/5a563710882ea4a7c7c201e1e99d1eb7b32c7572

**Comment by oliversalzburg** [comment]
@Mattijah I guess it should. I don't see how it doesn't affect the whole API in its current state. That being said, we're actually moving to a different translation approach in our own application, so I can't really help much.

**Comment by Mattijah** [comment]
@oliversalzburg if you don't initialize i18n with `req` and you use straight away e.g `i18n.__({phrase: "hello", locale: "en"})`, then it won't work. Never mind, thanks anyway

**Comment by grd2345** [comment]
How are things going here?

**Comment by mashpie** [comment]
Thank you sir :)

merged into develop - to be published with 0.6.0

**Comment by oliversalzburg** [comment]
Nice. Although I have recently fixed an issue with this feature in my fork.
I might be able to work on a PR later.

**Pull request closed without merging**